src/console_window.cpp

```diff
@@ -379,9 +396,78 @@ void ConsoleWindow::excludeFilterUpdated(const QString &text)
 
   db_proxy_->setExcludeFilters(filtered);
   db_proxy_->setExcludeRegexpPattern(text);
+  db_proxy_->clearSearchFailure();//resets failed search variables, VCM 27 April 2017
   updateExcludeLabel();
 }
 
+//Slot called when 'Search' text modified, 13 April 2017 VCM
+void ConsoleWindow::searchFilterUpdated(const QString &text)
+{
+	ConsoleWindow::updateSearchIndex("search");
+}
+//Slot called when 'Previous' button pushed, 13 April 2017 VCM
+void ConsoleWindow::on_pushPrev_clicked()
+{
+	ConsoleWindow::updateSearchIndex("prev");
+}
+//Slot called when 'Next' button pushed,  13 April 2017 VCM
+void ConsoleWindow::on_pushNext_clicked()
+{
+	ConsoleWindow::updateSearchIndex("next");
+}
+
+//        Search Function Definitions:
+//          1)search - user modified 'Search' text
+//          2)next   - user pressed 'Next' button
+//          3)prev   - user pressed 'Previous' button
+//			Locates and selects the next item based on search criteria, VCM 26 April 2017
+void ConsoleWindow::updateSearchIndex(QString function)
```

Comment:
If this can only possibly be three things, consider making this an `enum` instead of a `QString`.